packages/nextjs/src/common/utils/withIsolationScopeOrReuseFromRootSpan.ts

```diff
@@ -0,0 +1,39 @@
+import {
+  getActiveSpan,
+  getCapturedScopesOnSpan,
+  getDefaultIsolationScope,
+  getRootSpan,
+  spanToJSON,
+  withIsolationScope,
+} from '@sentry/core';
+import type { Scope } from '@sentry/types';
+
+/**
+ * Wrap a callback with a new isolation scope.
+ * However, if we have an active root span that was generated by next, we want to reuse the isolation scope from that span.
+ */
+export function withIsolationScopeOrReuseFromRootSpan<T>(cb: (isolationScope: Scope) => T): T {
+  const activeSpan = getActiveSpan();
+
+  if (!activeSpan) {
+    return withIsolationScope(cb);
+  }
+
+  const rootSpan = getRootSpan(activeSpan);
+
+  // Verify this is a next span
+  if (!spanToJSON(rootSpan).data?.['next.route']) {
```

Comment:
this feels like a reasonable starting point here I guess? We should sync this with the (not yet existing) isolation scope integration!

Reply:
We could also use `next.span_name` here as this attribute is added to (I think) almost all spans. Look [here](https://github.com/vercel/next.js/blob/canary/test/e2e/opentelemetry/opentelemetry.test.ts)